ff/src/biginteger/macros.rs

```diff
@@ -352,5 +352,36 @@ macro_rules! bigint_impl {
                 repr
             }
         }
+
+        impl From<BigUint> for $name {
+            #[inline]
+            fn from(val: num_bigint::BigUint) -> $name {
+                let bytes = val.to_bytes_le();
+                assert!(bytes.len() <= $num_limbs * 8);
+
+                let mut limbs = [0u64; $num_limbs];
+
+                bytes
+                    .chunks(8)
+                    .into_iter()
+                    .enumerate()
+                    .for_each(|(i, chunk)| {
+                        let mut chunk_padded = [0u8; 8];
+                        for j in 0..chunk.len() {
+                            chunk_padded[j] = chunk[j];
+                        }
+                        limbs[i] = u64::from_le_bytes(chunk_padded)
+                    });
+
+                Self(limbs)
+            }
+        }
```

Comment:
Maybe we change this to `TryFrom`? In case the `BigUInt is bigger?

Reply:
Done.

Reply:
(more fixings needed. on the way)